src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java

```diff
@@ -273,7 +273,20 @@ public final void checkContextAvailability(StepContext c) throws MissingContextV
             @SuppressFBWarnings(value="NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE", justification="all() will not return nulls")
             @Override
             public boolean apply(StepDescriptor i) {
-                return i.isMetaStep();
+                if (i.isMetaStep()) {
+                    Class<?> a = i.getMetaStepArgumentType();
+                    if (a != null) {
+                        if (!a.equals(Object.class)) {
+                            return true;
+                        } else {
```

Comment:
Confusing inverse-`if`.

Reply:
Yeah, made more sense in an earlier iteration. =)